### PR TITLE
Chore: Enable Merge HW controlled neurons

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Users can merge neurons controlled by HWs.

# Changes

* Check HW version in mergeNeurons service.
* Change in isMergeableNeuron util, check whether neuron is controllable instead checking if it's HW controlled.
* Remove case of HW controlled neuron in getMergeableNeuronMessageKey.

# Tests

* Change test case that allows merging HW controlled neurons.
* Add new test to check that mergeNeurons checks the HW version.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@mstrasinskis @peterpeterparker please review

File: frontend/src/lib/utils/neuron.utils.ts [hunk 1]
```diff
@@ -466,9 +465,6 @@ const getMergeableNeuronMessageKey = ({
   if (isSpawning(neuron)) {
     return "neurons.cannot_merge_neuron_spawning";
   }
-  if (isNeuronControlledByHardwareWallet({ neuron, accounts })) {
-    return "neurons.cannot_merge_hardware_wallet";
```
Comment (peterpeterparker) [inline frontend/src/lib/utils/neuron.utils.ts:470]: you can clean the i18n key

File: frontend/src/tests/lib/services/neurons.services.spec.ts [hunk 2]
```diff
@@ -815,6 +818,52 @@ describe("neurons-services", () => {
       expect(toastsShow).toHaveBeenCalled();
       expect(spyMergeNeurons).not.toHaveBeenCalled();
     });
+
+    it("should not update neuron if lower HW version than required", async () => {
```
Comment (peterpeterparker) [inline frontend/src/tests/lib/services/neurons.services.spec.ts:822]: typo? "should not update ..." shouldn't be "should not merge..."

File: frontend/src/lib/services/neurons.services.ts [hunk 3]
```diff
@@ -409,9 +411,17 @@ export const mergeNeurons = async ({
         translate({ labelKey: messageKey ?? "error.governance_error" })
       );
     }
+
     const identity: Identity = await getIdentityOfControllerByNeuronId(
       targetNeuronId
     );
+    const accounts = get(accountsStore);
+    if (isNeuronControlledByHardwareWallet({ neuron: neuron2, accounts })) {
```
Comment (peterpeterparker) [inline frontend/src/lib/services/neurons.services.ts:419]: should we test also if `neuron1` is hw?

File: frontend/src/lib/utils/neuron.utils.ts [hunk 4]
```diff
@@ -450,8 +450,7 @@ const isMergeableNeuron = ({
 }): boolean =>
   !hasJoinedCommunityFund(neuron) &&
   !isSpawning(neuron) &&
-  // Merging hardware wallet neurons is not yet supported
-  isNeuronControllableByUser({ neuron, mainAccount: accounts.main });
+  isNeuronControllable({ neuron, accounts });
```
Comment (peterpeterparker) [inline frontend/src/lib/utils/neuron.utils.ts:453]: should we test this or already covered by one?

**Review by peterpeterparker — APPROVED** [review]
Looks good, few minor questions so already approving

File: frontend/src/lib/services/neurons.services.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/lib/services/neurons.services.ts:419]: To be mergeable both need to have the same controller.

"neuron2" is the one where we call "manage_neuron" on. Which is the why we use the identity from.

But I'll rename `neuron1` and `neuron2` to `sourceNeuron` and `targetNeuron` to be clear.

File: frontend/src/lib/utils/neuron.utils.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/lib/utils/neuron.utils.ts:453]: It's the test I changed from `does not allow user to select neurons` to `allows user to select neurons`.

It's inside the `when mergeable neurons by hardware wallet`.

The test to check that when the controller is no account was already in place.

File: frontend/src/lib/utils/neuron.utils.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/utils/neuron.utils.ts:470]: thanks yes!

File: frontend/src/tests/lib/services/neurons.services.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/neurons.services.spec.ts:822]: yes, thanks!

File: frontend/src/lib/services/neurons.services.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/lib/services/neurons.services.ts]: and btw, the check whether they are the same controller is above in the function `canBeMerged`.

Comment (peterpeterparker) [reply frontend/src/lib/services/neurons.services.ts]: Gotcha, thanks

File: frontend/src/lib/utils/neuron.utils.ts (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/neuron.utils.ts:453]: Sounds good, thx

**Pull request merged**